api/novelist.py

```diff
@@ -433,6 +460,73 @@ def get_recommendations(self, metadata, recommendations_info):
                 metadata.recommendations += self._extract_isbns(book_info)
         return metadata
 
+    def get_isbns_from_query(self, library):
+        collectionList = []
+        for c in library.collections:
+            collectionList.append(c.id)
+
+        LEFT_OUTER_JOIN = True
+        i1 = aliased(Identifier)
+        i2 = aliased(Identifier)
+        e = aliased(Equivalency)
+        lp = aliased(LicensePool)
```

Comment:
This is a minor point but you don't need to alias Equivalency or LicensePool, since you're only using those tables once.